gstvstaudioprocessor.cpp

```diff
@@ -1137,15 +1151,116 @@ create_element_name(const gchar * prefix, const gchar * class_name)
   return element_name;
 }
 
+#if defined(__linux__)
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  gst_plugin_add_dependency_simple (plugin, NULL, "/usr/lib/vst3:/usr/local/lib/vst3", ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+  gst_plugin_add_dependency_simple (plugin, "HOME/.vst3", NULL, ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+}
+#elif defined(G_OS_WIN32)
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  PWSTR wideStr {};
+  char commonpath[MAX_PATH];
+  gchar *path;
+  if (FAILED (SHGetKnownFolderPath (FOLDERID_ProgramFilesCommon, 0, NULL, &wideStr)))
+    return;
+
+  if (!WideCharToMultiByte(CP_ACP, 0, wideStr, -1, commonpath, MAX_PATH, NULL, NULL))
+    return;
+
+  path = g_build_filename (commonpath, "VST3", NULL);
+  gst_plugin_add_dependency_simple (plugin, NULL, path, ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+  g_free (path);
+}
+#else
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  GST_FIXME_OBJECT (plugin, "Implement plugin dependencies support for this platform");
+}
+#endif
+
+static void
+list_files_with_extension (VST3::Hosting::Module::PathList &paths, const gchar *path, gboolean recurse)
+{
+  GDir *dir;
+  const gchar *dirent;
+  gchar *filename;
+
+  dir = g_dir_open (path, 0, NULL);
+  if (!dir)
+    return;
+
+  while ((dirent = g_dir_read_name (dir))) {
+    GStatBuf file_status;
+
+    filename = g_build_filename (path, dirent, NULL);
+    if (g_stat (filename, &file_status) < 0) {
+      g_free (filename);
+      continue;
+    }
+
+    if ((file_status.st_mode & S_IFDIR)) {
+      if (g_str_has_suffix (dirent, ".vst3"))
+        paths.push_back (filename);
```

Comment:
I think this is wrong on Windows? On Windows it's a *file* not a directory

Reply:
Yes that is correct, I took that out of the S_IFDIR check altogether as that is what VST3 does

Reply:
Yes, best to replicate the exact behaviour from module.cpp / module_{linux,win32}.cpp :)